src/Core/Framework/Api/Sync/SyncService.php

```diff
@@ -52,13 +52,17 @@ public function __construct(
     public function sync(array $operations, Context $context, SyncBehavior $behavior): SyncResult
     {
         if (\count($behavior->getSkipIndexers())) {
+            $context = clone $context;
```

Comment:
why not outside the if and once?

Reply:
No need to clone if no write happens

Reply:
when both happens you clone two times

Reply:
You want me to keep a bool flag that keeps the info whether I cloned already?